pydy/viz/server.py

```diff
@@ -2,36 +2,112 @@
 
 import os
 import sys
+import signal
+import socket
 import webbrowser
+
+# For python 2 and python 3 compatibility
 if sys.version_info < (3, 0):
     from SimpleHTTPServer import SimpleHTTPRequestHandler
     from BaseHTTPServer import HTTPServer
 else:
     from http.server import SimpleHTTPRequestHandler
     from http.server import HTTPServer
+    raw_input = input
+
+__all__ = ['Server']
+
+
+class StoppableHTTPServer(HTTPServer):
+    """
+    Overrides BaseHTTPServer.HTTPServer to include a stop
+    function.
+    """
+
+    def server_bind(self):
+        HTTPServer.server_bind(self)
+        self.socket.settimeout(1)
+        self.run = True
+
+    def get_request(self):
+        while self.run:
+            try:
+                sock, addr = self.socket.accept()
+                sock.settimeout(None)
+                return (sock, addr)
+            except socket.timeout:
+                pass
+
+    def stop(self):
+        self.run = False
+
+    def serve(self):
+        while self.run:
+            try:
+                self.handle_request()
+            except TypeError:
+                # When server is being closed, while loop can run once
+                # after setting self.run = False depending on how it
+                # is scheduled.
+                pass
+
 
+class Server(object):
+    """
+    Parameters
+    ----------
+    port : integer
+        Defines the port on which the server will run.
+    scene_file : name of the scene_file generated for visualization
+        Used here to display the url
+    directory : path of the directory which contains static and scene files.
+        Server is started in this directory itself.
 
-__all__ = ['run_server']
+    Example
+    -------
+    >>> server = Server(scene_file=_scene_json_file)
+    >>> server.run_server()
 
+    """
+    def __init__(self, port=8000, scene_file="Null"):
```

Comment:
Seems to me like it is a requirement. What use is the server if I don't have a scene file? I guess you could load it from a url `?scene_file=...` after running the server. If you want to support that use case, then you can make it a kwarg and `=None` or `=`""` but not `="Null"`. I'd prefer `None` as it follows common python patterns.